### ISSUE TITLE
Failed to execute 'postMessage' on 'Worker': #<Memory> could not be cloned.

### ISSUE THREAD

**Issue by jruiseco** [GoogleChromeLabs/wasm-bindgen-rayon#2]
Would it be possible to get some guidance on the above error from chrome?

1) Latest version of chrome
2) Using latest nightly with atomics and bulk-memory enabled
3) On latest version of the published crate

I am happy to continue chasing this down, but would appreciate a shove in the right direction. I am wondering if I missed a step in the setting features, perhaps the bulk-memory. Which is stopping the clone from working.

edit: Looks like there may be an issue with my cross origin isolation.

**Comment by RReverser** [comment]
Oh yeah, sorry I missed your issue. It does sound like you didn't have COOP/COEP headers set - did that fix your problem now?

**Comment by josephrocca** [comment]
I had this problem, and fixed it by setting [COOP/COEP headers](https://web.dev/coop-coep/) on the HTML file.

You'll also get this error if you skip the [Using Config Files](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon#using-config-files) section of the readme (i.e. set rust toolchain via `rust-toolchain` file and set flags via `.cargo/config`). @RReverser It might be worth moving those two instructions to the "Setting up" section to make it more clear?

In case it's helpful to others debugging this issue, [this deno script](https://github.com/josephrocca/denoSimpleStatic/blob/master/main.ts) served the files in the current directory with added COOP/COEP headers. You can use it like this (after [installing deno](https://deno.land/manual/getting_started/installation) of course):
```bash
deno run --allow-net --allow-read=. https://raw.githubusercontent.com/josephrocca/denoSimpleStatic/master/main.ts --port=3001
```

**Comment by jruiseco** [comment]
Definitely fixed with headers. Thank you.

**Comment by RReverser** [comment]
> @RReverser It might be worth moving those two instructions to the "Setting up" section to make it more clear?

Interesting idea, maybe. Although the line between "Setting up" and "Building Rust code" becomes kinda blurred - e.g. I assume `rust-toolchain` would also go to "Setting up" now, but then alternative usage via env vars would remain in "Building Rust code"?

**Comment by josephrocca** [comment]
@RReverser maybe I'm the only one who rushed through the readme and assumed that I was good to go after "Setting up". I assumed that was all the setup that I'd need to do, and so it took me a little while to work out that I hadn't completed all the steps.

Another "fix" for this could just be to somehow throw an error if they haven't completed the `rust-toolchain` and `.cargo/config` steps (if that's possible). Or maybe renaming the headers with `Step 1: ...`, `Step 2: ...`, `Step 3: ...` so it's clear that there are more things to do after the "Setting up" step.

**Comment by RReverser** [comment]
> maybe I'm the only one who rushed through the readme and assumed that I was good to go after "Setting up"

Yeah as I said I'm happy to improve docs in any way to make them clearer - I know it's a lot of setup, as mentioned in the beginning of README - just unsure of "how" :)

I thought having TOC which mentions all the steps in the beginning would be sufficient to see that they're meant to be executed one by one, where "Setting up" is just literally initial setup, followed by actually "Building Rust code".

**Comment by RReverser** [comment]
> Another "fix" for this could just be to somehow throw an error if they haven't completed the rust-toolchain and .cargo/config steps (if that's possible).

Unfortunately, I'm not sure this is possible at build-time - it only affects binary contents of the generated Wasm (whether memory is marked shared or not), and you can't inspect the final artifact from Rust itself, it only has `build.rs` for pre-building steps.

The only thing that could be possible is to improve the runtime message so that it detects when memory isn't shared and throws a clearer error in that case - would it help? If so, any chance you'd want to make a PR?

**Comment by josephrocca** [comment]
> If so, any chance you'd want to make a PR?

@RReverser Done, and also made a PR to help warn about the potentially-missing `no-bundler` feature from #3 👍

**Comment by RReverser** [comment]
Published as 1.0.3.
